components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -388,17 +388,39 @@ export class PrebuildManager {
         }
     }
 
-    private async shouldRateLimitPrebuild(span: opentracing.Span, cloneURL: string): Promise<boolean> {
+    private async shouldRateLimitPrebuild(span: opentracing.Span, cloneURL: string, userId: string): Promise<boolean> {
         const windowStart = secondsBefore(new Date().toISOString(), PREBUILD_LIMITER_WINDOW_SECONDS);
+
+        // per cloneULR rate limit
         const unabortedCount = await this.workspaceDB
             .trace({ span })
-            .countUnabortedPrebuildsSince(cloneURL, new Date(windowStart));
+            .countUnabortedPrebuildsPerCloneURLSince(cloneURL, new Date(windowStart));
         const limit = this.getPrebuildRateLimitForCloneURL(cloneURL);
 
         if (unabortedCount >= limit) {
-            log.debug("Prebuild exceeds rate limit", { limit, unabortedPrebuildsCount: unabortedCount, cloneURL });
+            log.debug({ userId }, "Prebuild exceeds per-cloneURL rate limit", {
+                limit,
+                unabortedPrebuildsCount: unabortedCount,
+                cloneURL,
+            });
             return true;
         }
+
+        // per user rate limit
+        const perUserRateLimit = this.config.prebuildPerUserRateLimit;
+        if (perUserRateLimit) {
```

Comment:
Do you want to also check that `perUserRateLimit` is not 0 or negative? What's the expected behaviour when the rate limit is 0? Code suggests it will allow unlimited, is that desired?